### ISSUE TITLE
decorated functions that return `None` aren't warned against

### ISSUE THREAD

**Issue by KotlinIsland** [KotlinIsland/basedmypy#569]
```py
from typing import overload

def f(o: object) -> None: ...

@overload
def o() -> None:...

@overload
def o(o: object) -> None:...

def o(o: object=1) -> None:...

a = f(1)  # error: 'f' does not return a value
b = o(1)
```
